source/WindowScroller/utils/onScroll.js

```diff
@@ -7,7 +7,13 @@ import {
 } from '../../utils/requestAnimationTimeout';
 import type WindowScroller from '../WindowScroller.js';
 
-let mountedInstances = [];
+// Checking a component's props is unreliable. Instead we store the element used at the time
```

Comment:
Why is this? Props change before unregistering?

Reply:
Yeah, we call this from `componentDidUpdate` and as a result any time we do the checks in `unregisterScrollListener` and `registerScrollListener` we always compare the new props against the given element which makes most of the calls fail